dev/NumberBox/NumberBoxAutomationPeer.cpp

```diff
@@ -31,6 +31,26 @@ hstring NumberBoxAutomationPeer::GetClassNameCore()
     return winrt::hstring_name_of<winrt::NumberBox>();
 }
 
+hstring NumberBoxAutomationPeer::GetNameCore()
+{
+    winrt::hstring name = __super::GetNameCore();
+
+    if (name.empty())
+    {
+        if (auto numberBox = Owner().try_as<winrt::NumberBox>())
+        {
+            name = SharedHelpers::TryGetStringRepresentationFromObject(numberBox.Header());
+        }
+
+        if (name.empty())
+        {
+            name = L"NumberBox";
```

Comment:
>NumberBox [](start = 21, length = 9)

Type().Name

Reply:
I can do winrt::get_class_name(numberBox), and that works, but it returns "Microsoft.UI.Xaml.Control.NumberBox" which feels kind of weird... Actually, I think maybe I should just leave it blank? The only controls we have which set a default name like this are items, not the parent control.